pkg/realizer/deliverable/component.go

```diff
@@ -84,9 +89,13 @@ func (r *resourceRealizer) Do(ctx context.Context, resource *v1alpha1.ClusterDel
 		templatingContext["source"] = inputs.OnlySource()
 	}
 
+	// TODO: Could we be leaking secrets? Should we just print the keys
+	//log.V(logger.DEBUG).Info("using context for stamper", "context", templatingContext)
```

Comment:
zap does provide a way of providing a custom marshaller for a given type

https://github.com/uber-go/zap/blob/eaeb0fc72fd23af7969c9a9f39e51b66827507ca/zapcore/marshaler.go#L23-L32

but, I'm not really sure we'd want to go that route - I particularly prefer the approach you mentioned (an extra level where you intentionally specify that you want this kind of logging)

Reply:
apparently kubernetes does provide some sanitization for their own logs? https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/component-base/logs/sanitization/sanitization.go

Reply:
Ive created a card on the board to do this for templatingcontext and outputs